ui/app/controllers/exec.js

```diff
@@ -35,14 +43,53 @@ export default Controller.extend({
     this.terminal.writeln('Select a task to start your session.');
   },
 
+  allocations: alias('model.allocations'),
+
+  taskState: computed(
+    'allocations.[]',
+    'allocationShortId',
+    'allocations.@each.isActive',
+    'taskName',
+    'taskGroupName',
+    'allocation',
+    'allocation.states.@each.name',
+    'allocation.states.@each.isRunning',
```

Comment:
Lol, so many dependent keys. Gotta love our data model.

Reply:
I had an intermediate `allocation` computed property that would have reduced this somewhat but it didn’t work for whatever reason and I gave up on it  😞